vm/vm.go

```diff
@@ -58,19 +58,47 @@ const (
 	MinAcceptedBlockWindow = 1024
 )
 
+var _ Option = (*OptionFunc)(nil)
+
+type Options struct {
+	BlockBuilder builder.Builder
+	TxGossiper   gossiper.Gossiper
+}
```

Comment:
Could we store these values directly on the VM instead of in an options struct or is this style the standard? Since `Options` has direct access to the `*VM` type it seems like we can.

Going forward, when should we use an option as opposed to an argument to the constructor? I assume the constructor takes in the values every VM needs to provide and options enable configuring anything that is more custom.

Reply:
Should options be moved to a separate file?

Reply:
I re-implemented... let me know if you think this looks better. 

> Going forward, when should we use an option as opposed to an argument to the constructor? I assume the constructor takes in the values every VM needs to provide and options enable configuring anything that is more custom.

I agree with this

> Should options be moved to a separate file?

Moved... I think we do want to probably delete the options I added though in the future when we finish merging https://github.com/ava-labs/hypersdk/pull/1195.